### ISSUE TITLE
Fan adjustments based off CPU temps

### ISSUE THREAD

**Issue by noahclements** [Danielv123/serverManager#29]
Have it running on my R720 with iDRAC 8 at the moment. 

It would be nice to have some more options for fan control based off of CPU temperatures rather than solely ambient temp. This would solve my current issue of doing an intensive task on the server and CPU's spiking to ~80 degrees C while fans remain at ~2600RPM.

This repo aggregates multiple sensors to base the fan speeds off of: https://github.com/spacelama/R710-Fan-Control

**Comment by Danielv123** [comment]
Hi. It should be going after the highest reported temperature, which is generally the hottest CPU.Are you sure it's not just that it responds slowly?On Feb 8, 2022 21:39, Noah Clements ***@***.***> wrote:
Have it running on my R720 with iDRAC 8 at the moment.
It would be nice to have some more options for fan control based off of CPU temperatures rather than solely ambient temp. This would solve my current issue of doing an intensive task on the server and CPU's spiking to ~80 degrees C while fans remain at ~2600RPM.
This repo aggregates multiple sensors to base the fan speeds off of: https://github.com/spacelama/R710-Fan-Control

**Comment by noahclements** [comment]
After doing some testing, I think the issue instead revolves around the data update halting after what seems like one ipmi call (Very similar to #2). I wasn't even able to see my server's info until i restarted the container to begin with, which then created a 2nd tab labelled "Dell" that included the info. 

After that first ipmitool call, it tends to hang and the fan RPM does not stick to the fan control curve.

**Comment by ihatemyisp** [comment]
> After doing some testing, I think the issue instead revolves around the data update halting after what seems like one ipmi call (Very similar to #2). I wasn't even able to see my server's info until i restarted the container to begin with, which then created a 2nd tab labelled "Dell" that included the info.
> 
> After that first ipmitool call, it tends to hang and the fan RPM does not stick to the fan control curve.

What you might try is manually creating/fixing your servers.json with the container shutdown. While I haven't had played with the fan curve at all, I can vouch that everything else is stable once the servers.json is squared away.

```
[
    {
        "name": "R720",
        "address": "<ip/host>",
        "username": "<username>",
        "password": "",
        "sensordataRaw": [],
        "sensordata": []
    }
]
```

**Comment by noahclements** [comment]
Manually wiping servers.json and pasting that back in w/ my info before starting the container worked! Looks to be updating regularly now, thank you!

**Comment by Danielv123** [comment]
I believe this should have been fixed in the latest version of the image. Please reopen if that is not the case.
